docs/rules-engine/how-to-use/industrialize-ruleset-generation.md

```diff
@@ -0,0 +1,297 @@
+    },
+    {
+      "name": "fact3",
+      "description": "Description of fact3",
+      "type": "string",
+      "enum": ["a", "b"]
+    }
+  ]
+}
+```
+
+You can note that the ``ComplexInterface`` fact results in a separated schema file that validates the interface model:
+
+```json5 
+{
+  "type": "object",
+  "properties": {
+    "property1": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/OtherInterface"
+      }
+    }
+  },
+  "definitions": {
+    "OtherInterface": {
+      "type": "object",
+      "description": "Description of property1",
+      "properties": {
+        "property": {
+          "description": "Description of property",
+          "type": "number"
+        }
+      }
+    }
+  }
+}
+```
+
+> [!NOTE]
+>
+> The models for the OperatorsMetadata is available in the [rules engine metadata interface file](https://github.com/AmadeusITGroup/otter/blob/main/packages/%40o3r/rules-engine/builders/rules-engine-extractor/helpers/rules-engine.extractor.interfaces.ts).
+
+### Project with module subentries
+
+As mentioned in the previous section, the extractor looks into your project and its node_modules files to identify the
+schema of your complex facts.
+
+If your project is a library with sub-entries, the schema generator will not know how to find the files they contain as
+they are not part of your module index.ts nor part of your node_modules. 
+If you define your fact based on interfaces defined in your sub-entries, it will result in an incomplete schema.
+
+If you find yourself in this situation, you can specifically include additional paths to the sub-entries in the extractor
+[tsconfig file](#how-to-use-it).
+
+```JSON5
+// tsconfig.cms.json
+{
+  // ...
+  "extraOptions": {
+    "otterSubModuleRefs": ["./**/my-sub-entry/index.ts"], // Relative path to the index.ts where the interfaces of your sub-entry are exported
+  },
+  // ...
+}
+```
+
+This will allow the extractor to find the definition of your sub-entry interface and generate their schema files.
+For instance, let's consider the MyFacts object defined below.
+
+```typescript
+// myFacts.ts
+import {SubEntryClass} from '@myLib/my-sub-entry/subEntry';
+
+export interface MyFacts extends FactDefinitions {
+  someProperty: SubEntryClass;
+}
+```
+
+The extraction will result in an object with a reference to the SubEntryClass model which will be created during the extraction:
+
+```json5
+{
+  "facts": [
+    {
+      "name": "someProperty",
+      "type": "object",
+      "schemaFile": "fact-schemas/sub-entry.schema.json"
+    }
+  ]
+}
+```
+
+> [!WARNING]
+> 
+> The fact extractor is not able to make references to sub-entry schema in other model schema files. 
+> That means that if the ``SubEntryClass`` refers to another sub-entry it will be ignored.
+> 
+> In the following example, there will be a schema generated for the SubEntryClass, and it will describe ``SameSubEntryClass``
```

Comment:
```suggestion
> In the following example, there will be a schema generated for the ``SubEntryClass``, and it will describe the definitions of ``SameSubEntryClass`` and ``SomeNodeModulesRef``, but will ignore the ``OtherSubEntryClass`` definition.
```